Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/CrashesContentPage.xaml.cs

```diff
@@ -25,6 +32,59 @@ public CrashesContentPage()
             base.OnAppearing();
             CrashesEnabledSwitchCell.On = await Crashes.IsEnabledAsync();
             CrashesEnabledSwitchCell.IsEnabled = await AppCenter.IsEnabledAsync();
+
+            // Attachments
+            if (Application.Current.Properties.TryGetValue(TextAttachmentKey, out var textAttachment) &&
+                textAttachment is string text)
+            {
+                TextAttachmentCell.Detail = text;
+            }
+            if (Application.Current.Properties.TryGetValue(FileAttachmentKey, out var fileAttachment) &&
+                fileAttachment is string file)
+            {
+                var filePicker = DependencyService.Get<IFilePicker>();
+                try
+                {
+                    FileAttachmentCell.Detail = filePicker?.GetFileDescription(file);
+                }
+                catch (Exception e)
+                {
+                    Debug.WriteLine("Couldn't read file attachment: {0}", e.Message);
+                    Application.Current.Properties.Remove(FileAttachmentKey);
+                }
+            }
+            if (XamarinDevice.RuntimePlatform == XamarinDevice.UWP)
+            {
+                TextAttachmentCell.IsEnabled = false;
+                FileAttachmentCell.IsEnabled = false;
+            }
+        }
+
+        async void UpdateEnabled(object sender, ToggledEventArgs e)
+        {
+            await Crashes.SetEnabledAsync(e.Value);
+        }
+
+        async void TextAttachment(object sender, EventArgs e)
+        {
+            var text = await TextAttachmentView.Show(Navigation);
+            ((TextCell)sender).Detail = text;
+            Application.Current.Properties[TextAttachmentKey] = text;
+            await Application.Current.SavePropertiesAsync();
+        }
+
+        async void FileAttachment(object sender, EventArgs e)
+        {
+            var filePicker = DependencyService.Get<IFilePicker>();
+            if (filePicker == null)
+            {
+                Debug.WriteLine("File attachment isn't implemented");
+                return;
+            }
+            var file = await filePicker.PickFile();
```

Comment:
When I run the app in simulator, I get a crash at this line

Reply:
@achocron What OS? What crash message?

Reply:
iPhone 8 iOS 11.1. It's a native crash so there's no .NET exception